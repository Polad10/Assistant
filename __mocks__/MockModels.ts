const appointment_1 = { id: 1, datetime: '2024-01-01 19:44:00.000Z', actions: null, treatment_id: 1 }
const appointment_2 = { id: 2, datetime: '2024-01-02 19:44:00.000Z', actions: null, treatment_id: 2 }

const patient_1 = {
  id: 1,
  first_name: 'FirstName',
  last_name: 'LastName',
  dob: null,
  city: null,
  phone: null,
  extra_info: null,
}

const patient_2 = {
  id: 2,
  first_name: 'FirstName2',
  last_name: 'LastName2',
  dob: null,
  city: null,
  phone: null,
  extra_info: null,
}

const payment_1 = {
  id: 1,
  date: '2024-01-01 19:44:00.000Z',
  amount: 100,
  treatment_id: 1,
}

const payment_2 = {
  id: 2,
  date: '2024-01-02 19:44:00.000Z',
  amount: 100,
  treatment_id: 2,
}

const treatment_1 = {
  id: 1,
  start_date: '2024-01-01 19:44:00.000Z',
  end_date: null,
  title: 'Title',
  patient_id: 1,
  price: 100,
}

const treatment_2 = {
  id: 2,
  start_date: '2024-01-02 19:44:00.000Z',
  end_date: null,
  title: 'Title2',
  patient_id: 2,
  price: 100,
}

const setting = {
  language: 'en',
}

export const mockModels = {
  appointment_1,
  appointment_2,
  patient_1,
  patient_2,
  payment_1,
  payment_2,
  treatment_1,
  treatment_2,
  setting,
}
