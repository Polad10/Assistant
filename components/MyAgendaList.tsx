import { useCallback, useContext, useEffect, useState } from 'react'
import { AgendaList, AgendaListProps } from 'react-native-calendars'
import AgendaItem from './AgendaItem'
import MainView from './MainView'
import { RootStackParamList } from '../types/Navigation'
import { DeviceEventEmitter, StyleSheet, View } from 'react-native'
import { Colors } from '../types/Colors'
import { ThemeContext, ThemeContextType } from '../contexts/ThemeContext'
import { Divider } from '@rneui/themed'
import LoadingView from './LoadingView'

interface Props extends AgendaListProps {
  pageName: keyof RootStackParamList
}

export default function MyAgendaList(props: Props) {
  const themeContext = useContext(ThemeContext)!
  const [loading, setLoading] = useState(false)

  useEffect(() => {
    const loadingListener = DeviceEventEmitter.addListener('loading', () => setLoading(true))
    const loadingFinishedListener = DeviceEventEmitter.addListener('loadingFinished', () => setLoading(false))

    return () => {
      loadingListener.remove()
      loadingFinishedListener.remove()
    }
  }, [])

  const renderItem = useCallback(({ item }: any) => {
    return (
      <View>
        <AgendaItem appointment={item} />
        <Divider color={themeContext.border} />
      </View>
    )
  }, [])

  function getContentView() {
    return (
      <>
        <AgendaList
          sections={props.sections}
          renderItem={renderItem}
          sectionStyle={styles(themeContext).agendaSection}
        />
        {loading && <LoadingView />}
      </>
    )
  }

  return <MainView>{getContentView()}</MainView>
}

const styles = (themeContext: ThemeContextType) =>
  StyleSheet.create({
    agendaSection: {
      backgroundColor: themeContext.secondary,
      color: themeContext.neutral,
    },
    divider: {
      marginHorizontal: 10,
    },
  })
