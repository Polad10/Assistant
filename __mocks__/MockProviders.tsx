import { Dispatch, PropsWithChildren, SetStateAction } from 'react'
import { Translator } from '../helpers/Translator'
import { AppointmentRequest } from '../modals/Appointment'
import { PatientRequest } from '../modals/Patient'
import { Setting } from '../modals/Setting'
import { TreatmentRequest } from '../modals/Treatment'
import { PaymentRequest } from '../modals/Payment'
import { Api } from '../helpers/Api'
import { mockModels } from './MockModels'
import { AuthContext } from '../contexts/AuthContext'
import { DataContext } from '../contexts/DataContext'
import { ThemeContext } from '../contexts/ThemeContext'
import { LocalizationContext } from '../contexts/LocalizationContext'

const themeContextMock = {
  primary: '#121212',
  secondary: '#1a1a1a',
  accent: '#7600eb',
  neutral: '#faf8ff',
  border: '#374151',
  info: '#d3fbd8',
  success: '#00e5b0',
  warning: '#9f1239',
  error: '#9f1239',
}

const localizationContextMock = {
  translator: new Translator('en'),
  language: 'en',
}

const authContextMock = {
  user: undefined,
  loading: false,
}

const dataContextMock = {
  fetchPatients: async () => {},
  fetchAppointments: async () => {},
  fetchTreatments: async () => {},
  fetchPayments: async () => {},
  fetchSetting: async () => {},

  createAppointment: async (appointmentRequest: AppointmentRequest) => mockModels.appointment_1,
  updateAppointment: async (appointmentRequest: AppointmentRequest) => mockModels.appointment_1,
  deleteAppointment: async (appointmentId: number) => {},

  createPatient: async (patientRequest: PatientRequest) => mockModels.patient_1,
  updatePatient: async (patientRequest: PatientRequest) => mockModels.patient_1,
  deletePatient: async (patientId: number) => {},

  createTreatment: async (treatmentRequest: TreatmentRequest) => mockModels.treatment_1,
  updateTreatment: async (treatmentRequest: TreatmentRequest) => mockModels.treatment_1,
  deleteTreatment: async (treatmentId: number) => {},

  createPayment: async (paymentRequest: PaymentRequest) => mockModels.payment_1,
  updatePayment: async (paymentRequest: PaymentRequest) => mockModels.payment_1,
  deletePayment: async (paymentId: number) => {},

  createSetting: async (setting: Setting) => {},
  updateSetting: async (setting: Setting) => {},

  api: undefined,
  setApi: (action: SetStateAction<Api | undefined>): Dispatch<SetStateAction<Api | undefined>> => jest.fn(),

  loading: false,
  setLoading: (action: SetStateAction<boolean>): Dispatch<SetStateAction<boolean>> => jest.fn(),

  patients: [mockModels.patient_1, mockModels.patient_2],
  appointments: [mockModels.appointment_1, mockModels.appointment_2],
  treatments: [mockModels.treatment_1, mockModels.treatment_2],
  payments: [mockModels.payment_1, mockModels.payment_2],
  setting: mockModels.setting,
}

export default function MockProviders({ children }: PropsWithChildren) {
  return (
    <AuthContext.Provider value={authContextMock}>
      <DataContext.Provider value={dataContextMock}>
        <ThemeContext.Provider value={themeContextMock}>
          <LocalizationContext.Provider value={localizationContextMock}>{children}</LocalizationContext.Provider>
        </ThemeContext.Provider>
      </DataContext.Provider>
    </AuthContext.Provider>
  )
}
