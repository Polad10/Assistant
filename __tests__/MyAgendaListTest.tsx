import renderer from 'react-test-renderer'
import MockProviders from '../__mocks__/MockProviders'
import MyAgendaList from '../components/MyAgendaList'
import { mockModels } from '../__mocks__/MockModels'

const sections = [
  { title: '2024-01-01', data: [mockModels.appointment_1] },
  { title: '2024-01-02', data: [mockModels.appointment_2] },
]

describe('MyAgendaList', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MockProviders>
          <MyAgendaList pageName='Appointments' sections={sections} />
        </MockProviders>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
