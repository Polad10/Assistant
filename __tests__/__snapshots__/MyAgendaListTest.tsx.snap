// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyAgendaList renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "backgroundColor": "#121212",
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RCTScrollView
    data={
      [
        {
          "data": [
            {
              "actions": null,
              "datetime": "2024-01-01 19:44:00.000Z",
              "id": 1,
              "status": 1,
              "treatment_id": 1,
            },
          ],
          "title": "2024-01-01",
        },
        {
          "data": [
            {
              "actions": null,
              "datetime": "2024-01-02 19:44:00.000Z",
              "id": 2,
              "status": 2,
              "treatment_id": 2,
            },
          ],
          "title": "2024-01-02",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    onScrollToIndexFailed={[Function]}
    onViewableItemsChanged={[Function]}
    renderItem={[Function]}
    scrollEventThrottle={50}
    sectionStyle={
      {
        "backgroundColor": "#1a1a1a",
        "color": "#faf8ff",
      }
    }
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={
      [
        0,
        3,
      ]
    }
    viewabilityConfig={
      {
        "itemVisiblePercentThreshold": 20,
      }
    }
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <Text
          allowFontScaling={false}
          onLayout={[Function]}
          style={
            [
              {
                "backgroundColor": "white",
                "color": "#7a92a5",
                "fontSize": 12,
                "fontWeight": "bold",
                "lineHeight": 16,
                "paddingBottom": 8,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 24,
                "textAlign": "left",
                "textTransform": "uppercase",
              },
              {
                "backgroundColor": "#1a1a1a",
                "color": "#faf8ff",
              },
            ]
          }
        >
          today, Monday, Jan 1
        </Text>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View>
          <View
            collapsable={false}
            handlerTag={1}
            handlerType="PanGestureHandler"
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            onLayout={[Function]}
            style={
              {
                "overflow": "hidden",
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "bottom": 0,
                  "flexDirection": "row-reverse",
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": [
                    {
                      "translateX": -10000,
                    },
                  ],
                }
              }
            >
              <View
                style={
                  {
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "#00e5b0",
                      "justifyContent": "center",
                      "opacity": 1,
                      "width": 80,
                    }
                  }
                >
                  <Text />
                  <Text
                    style={
                      {
                        "color": "#faf8ff",
                        "marginTop": 5,
                      }
                    }
                  >
                    Finish
                  </Text>
                </View>
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "#9f1239",
                      "justifyContent": "center",
                      "opacity": 1,
                      "width": 80,
                    }
                  }
                >
                  <Text />
                  <Text
                    style={
                      {
                        "color": "#faf8ff",
                        "marginTop": 5,
                      }
                    }
                  >
                    Cancel
                  </Text>
                </View>
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "#9f1239",
                      "justifyContent": "center",
                      "opacity": 1,
                      "width": 80,
                    }
                  }
                >
                  <Text />
                  <Text
                    style={
                      {
                        "color": "#faf8ff",
                        "marginTop": 5,
                      }
                    }
                  >
                    Delete
                  </Text>
                </View>
              </View>
              <View
                onLayout={[Function]}
              />
            </View>
            <View
              collapsable={false}
              handlerTag={2}
              handlerType="TapGestureHandler"
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              pointerEvents="auto"
              style={
                {
                  "transform": [
                    {
                      "translateX": -0,
                    },
                  ],
                }
              }
            >
              <View
                style={
                  {
                    "backgroundColor": "#121212",
                  }
                }
              >
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "backgroundColor": "#121212",
                      "flexDirection": "row",
                      "opacity": 1,
                      "padding": 10,
                    }
                  }
                >
                  <View
                    style={
                      {
                        "flex": 1,
                        "flexDirection": "row",
                      }
                    }
                  >
                    <View
                      style={
                        {
                          "width": 60,
                        }
                      }
                    >
                      <Text
                        style={
                          [
                            {
                              "color": "#faf8ff",
                              "fontSize": 16,
                            },
                            {
                              "color": "#faf8ff",
                              "fontWeight": "bold",
                            },
                            {
                              "opacity": 1,
                              "textDecorationLine": "none",
                            },
                          ]
                        }
                      >
                        Invalid DateTime
                      </Text>
                    </View>
                    <View
                      style={
                        {
                          "flex": 1,
                        }
                      }
                    >
                      <View>
                        <Text
                          style={
                            [
                              {
                                "color": "#faf8ff",
                                "fontSize": 16,
                              },
                              {
                                "color": "#d3fbd8",
                              },
                              {
                                "opacity": 1,
                                "textDecorationLine": "none",
                              },
                            ]
                          }
                        >
                          FirstName LastName
                        </Text>
                      </View>
                      <Text
                        style={
                          [
                            {
                              "color": "#faf8ff",
                              "fontSize": 16,
                            },
                            {
                              "fontStyle": "italic",
                              "marginTop": 10,
                            },
                            {
                              "opacity": 1,
                              "textDecorationLine": "none",
                            },
                          ]
                        }
                      />
                    </View>
                    <View
                      style={
                        {
                          "marginLeft": 10,
                          "width": 100,
                        }
                      }
                    >
                      <View
                        style={
                          {
                            "alignItems": "center",
                            "flexDirection": "row",
                          }
                        }
                      >
                        <View
                          style={{}}
                          testID="RNE__Badge__Container"
                        >
                          <View
                            style={
                              {
                                "alignItems": "center",
                                "alignSelf": "center",
                                "backgroundColor": undefined,
                                "borderColor": undefined,
                                "borderRadius": 4,
                                "borderWidth": 0.5,
                                "height": 8,
                                "justifyContent": "center",
                                "minWidth": 8,
                                "paddingHorizontal": 0,
                                "paddingVertical": 0,
                              }
                            }
                            testID="RNE__Badge"
                          />
                        </View>
                        <Text
                          style={
                            {
                              "color": "#faf8ff",
                              "marginLeft": 10,
                            }
                          }
                        >
                          Expected
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              {
                "borderBottomColor": "#374151",
                "borderBottomWidth": 0.5,
              }
            }
            testID="RNE__Divider"
            theme={
              {
                "colors": {
                  "background": "#ffffff",
                  "black": "#242424",
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": {
                    "android": {
                      "error": "#f44336",
                      "grey": "rgba(0, 0, 0, 0.54)",
                      "primary": "#2196f3",
                      "searchBg": "#dcdce1",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "default": {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "ios": {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "web": {
                      "error": "#ff190c",
                      "grey": "#393e42",
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#ca71eb",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ad1457",
                  "success": "#52c41a",
                  "warning": "#faad14",
                  "white": "#ffffff",
                },
                "spacing": {
                  "lg": 12,
                  "md": 8,
                  "sm": 4,
                  "xl": 24,
                  "xs": 2,
                },
              }
            }
          />
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      />
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <Text
          allowFontScaling={false}
          onLayout={[Function]}
          style={
            [
              {
                "backgroundColor": "white",
                "color": "#7a92a5",
                "fontSize": 12,
                "fontWeight": "bold",
                "lineHeight": 16,
                "paddingBottom": 8,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 24,
                "textAlign": "left",
                "textTransform": "uppercase",
              },
              {
                "backgroundColor": "#1a1a1a",
                "color": "#faf8ff",
              },
            ]
          }
        >
          Tuesday, Jan 2
        </Text>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View>
          <View
            collapsable={false}
            handlerTag={3}
            handlerType="PanGestureHandler"
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            onLayout={[Function]}
            style={
              {
                "overflow": "hidden",
              }
            }
          >
            <View
              collapsable={false}
              style={
                {
                  "bottom": 0,
                  "flexDirection": "row-reverse",
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": [
                    {
                      "translateX": -10000,
                    },
                  ],
                }
              }
            >
              <View
                style={
                  {
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "#1a1a1a",
                      "justifyContent": "center",
                      "opacity": 1,
                      "width": 80,
                    }
                  }
                >
                  <Text />
                  <Text
                    style={
                      {
                        "color": "#faf8ff",
                        "marginTop": 5,
                      }
                    }
                  >
                    Restore
                  </Text>
                </View>
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "#9f1239",
                      "justifyContent": "center",
                      "opacity": 1,
                      "width": 80,
                    }
                  }
                >
                  <Text />
                  <Text
                    style={
                      {
                        "color": "#faf8ff",
                        "marginTop": 5,
                      }
                    }
                  >
                    Delete
                  </Text>
                </View>
              </View>
              <View
                onLayout={[Function]}
              />
            </View>
            <View
              collapsable={false}
              handlerTag={4}
              handlerType="TapGestureHandler"
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              pointerEvents="auto"
              style={
                {
                  "transform": [
                    {
                      "translateX": -0,
                    },
                  ],
                }
              }
            >
              <View
                style={
                  {
                    "backgroundColor": "#121212",
                  }
                }
              >
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "backgroundColor": "#121212",
                      "flexDirection": "row",
                      "opacity": 1,
                      "padding": 10,
                    }
                  }
                >
                  <View
                    style={
                      {
                        "flex": 1,
                        "flexDirection": "row",
                      }
                    }
                  >
                    <View
                      style={
                        {
                          "width": 60,
                        }
                      }
                    >
                      <Text
                        style={
                          [
                            {
                              "color": "#faf8ff",
                              "fontSize": 16,
                            },
                            {
                              "color": "#faf8ff",
                              "fontWeight": "bold",
                            },
                            {
                              "opacity": 0.5,
                              "textDecorationLine": "line-through",
                            },
                          ]
                        }
                      >
                        Invalid DateTime
                      </Text>
                    </View>
                    <View
                      style={
                        {
                          "flex": 1,
                        }
                      }
                    >
                      <View>
                        <Text
                          style={
                            [
                              {
                                "color": "#faf8ff",
                                "fontSize": 16,
                              },
                              {
                                "color": "#d3fbd8",
                              },
                              {
                                "opacity": 0.5,
                                "textDecorationLine": "line-through",
                              },
                            ]
                          }
                        >
                          FirstName2 LastName2
                        </Text>
                      </View>
                      <Text
                        style={
                          [
                            {
                              "color": "#faf8ff",
                              "fontSize": 16,
                            },
                            {
                              "fontStyle": "italic",
                              "marginTop": 10,
                            },
                            {
                              "opacity": 0.5,
                              "textDecorationLine": "line-through",
                            },
                          ]
                        }
                      />
                    </View>
                    <View
                      style={
                        {
                          "marginLeft": 10,
                          "width": 100,
                        }
                      }
                    >
                      <View
                        style={
                          {
                            "alignItems": "center",
                            "flexDirection": "row",
                          }
                        }
                      >
                        <View
                          style={{}}
                          testID="RNE__Badge__Container"
                        >
                          <View
                            style={
                              {
                                "alignItems": "center",
                                "alignSelf": "center",
                                "backgroundColor": "#9f1239",
                                "borderColor": "#9f1239",
                                "borderRadius": 4,
                                "borderWidth": 0.5,
                                "height": 8,
                                "justifyContent": "center",
                                "minWidth": 8,
                                "paddingHorizontal": 0,
                                "paddingVertical": 0,
                              }
                            }
                            testID="RNE__Badge"
                          />
                        </View>
                        <Text
                          style={
                            {
                              "color": "#faf8ff",
                              "marginLeft": 10,
                            }
                          }
                        >
                          Cancelled
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              {
                "borderBottomColor": "#374151",
                "borderBottomWidth": 0.5,
              }
            }
            testID="RNE__Divider"
            theme={
              {
                "colors": {
                  "background": "#ffffff",
                  "black": "#242424",
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": {
                    "android": {
                      "error": "#f44336",
                      "grey": "rgba(0, 0, 0, 0.54)",
                      "primary": "#2196f3",
                      "searchBg": "#dcdce1",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "default": {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "ios": {
                      "error": "#ff3b30",
                      "grey": "#7d7d7d",
                      "primary": "#007aff",
                      "searchBg": "#dcdce1",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                    "web": {
                      "error": "#ff190c",
                      "grey": "#393e42",
                      "primary": "#2089dc",
                      "searchBg": "#303337",
                      "secondary": "#ca71eb",
                      "success": "#52c41a",
                      "warning": "#faad14",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ad1457",
                  "success": "#52c41a",
                  "warning": "#faad14",
                  "white": "#ffffff",
                },
                "spacing": {
                  "lg": 12,
                  "md": 8,
                  "sm": 4,
                  "xl": 24,
                  "xs": 2,
                },
              }
            }
          />
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      />
    </View>
  </RCTScrollView>
</View>
`;
